project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -16,7 +16,12 @@
 import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.Query.FilterOperator;
 import com.google.appengine.api.datastore.Query.FilterPredicate;
-
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import java.util.HashMap;
+import java.util.Map;
 
 /** Class containing basic database functionalities. */
 public class Database {
```

Comment:
For this class specifically I would recommend that we merge some of the earlier PRs into main first before this. Just because the structural changes made would have a hard time merging into this rather than changing this to merge into the new Database class.